libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -525,7 +526,10 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
             }
 
             // Check if user entered a custom domain
-            if (SalesforceSDKManager.getInstance().shouldInferCustomDomain() && isNewLoginUrl(uri)) {
+            String host = uri.getHost();
+            Pattern customDomainPattern = SalesforceSDKManager.getInstance().getCustomDomainInferencePattern();
+            if (host != null && !getLoginUrl().contains(host) && customDomainPattern != null
```

Comment:
Comparing with [iOS](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/3636/files#diff-fc38a1c173394406330d16308ea5eee16aabbbe120e879cf10650d93b4ebe613R780), should we have a flag like `domainUpdated`, or is that achieved by setting the login server which causes `getLoginUrl().contains(host)` to be true if you are back here a second time?